From 8fca90257341365b03d2b0eed448bf0cfca8a4e9 Mon Sep 17 00:00:00 2001
From: Andy Ross <andy.ross@windriver.com>
Date: Thu, 9 May 2013 11:39:18 -0700
Subject: Allow default video encoder to be set via properties

Some apps (and more than a few CTS cases) don't use the MediaProfiles
interface to choose encoders, and the hard-coded default of H.263 is a
poor fit for systems where only H.264 encode is accelerated by
hardware.

For: AXIA-2634
Change-Id: I3e4a1ec3266031c080853d90699aef1bf4d0805a
Signed-off-by: Andy Ross <andy.ross@windriver.com>
---
 .../libmediaplayerservice/StagefrightRecorder.cpp  |   15 ++++++++++++++-
 1 files changed, 14 insertions(+), 1 deletions(-)

diff --git a/media/libmediaplayerservice/StagefrightRecorder.cpp b/media/libmediaplayerservice/StagefrightRecorder.cpp
index b2cf810..9eb7e63 100644
--- a/media/libmediaplayerservice/StagefrightRecorder.cpp
+++ b/media/libmediaplayerservice/StagefrightRecorder.cpp
@@ -42,6 +42,7 @@
 #include <camera/ICamera.h>
 #include <camera/CameraParameters.h>
 #include <gui/Surface.h>
+#include <cutils/properties.h>
 
 #include <utils/Errors.h>
 #include <sys/types.h>
@@ -172,7 +173,19 @@ status_t StagefrightRecorder::setVideoEncoder(video_encoder ve) {
     }
 
     if (ve == VIDEO_ENCODER_DEFAULT) {
-        mVideoEncoder = VIDEO_ENCODER_H263;
+        char prop[PROPERTY_VALUE_MAX];
+        property_get("ro.video.encoder.default", prop, "");
+        if(strcmp(prop, "h263") == 0) {
+            mVideoEncoder = VIDEO_ENCODER_H263;
+        } else if(strcmp(prop, "h264") == 0) {
+            mVideoEncoder = VIDEO_ENCODER_H264;
+        } else if(strcmp(prop, "m4v") == 0) {
+            mVideoEncoder = VIDEO_ENCODER_MPEG_4_SP;
+        } else {
+            if(prop[0])
+                ALOGE("Invalid ro.video.encoder.default property, expect one of h263,h264,m4v\n");
+            mVideoEncoder = VIDEO_ENCODER_H263;
+        }
     } else {
         mVideoEncoder = ve;
     }
-- 
1.7.4.1

